### PR TITLE
Add support for lists

### PR DESCRIPTION
**Pull request by hellricer**
Both direct access & access through the user model are implemented.

### CODE REVIEW

**Comment by JOJ0** [comment]
To me this is looking very well thought through and complete. I didn't test the feature, just ran your test successfully. Let's wait for Alistairs review then.

File: discogs_client/models.py [hunk 1]
```diff
@@ -621,6 +624,24 @@ def __repr__(self):
...
 
 
+class List(PrimaryAPIObject):
+    id = SimpleField()
+    name = SimpleField()
+    description = SimpleField()
+    public = SimpleField()
+    url = SimpleField(key='uri')
+    date_changed = SimpleField(transform=parse_timestamp)
+    date_added = SimpleField(transform=parse_timestamp)
```
Comment (alifhughes) [inline discogs_client/models.py:634]: Only a very slight suggestion, could change this to `date_created` to be [more consistent with the API docs](https://www.discogs.com/developers#page:user-lists,header:user-lists-list)? Dated added/changed are _almost_ synonyms, however I do understand the difference so am happy for it to go through.

Comment (hellricer) [reply discogs_client/models.py:634]: Hmm, that's strange - if you request the list through API, there's `date_added` and `date_changed` (see https://api.discogs.com/lists/1).

I've never seen a response like the one in the snippet you linked. Gotta investigate more.

Comment (JOJ0) [reply discogs_client/models.py:634]: Yep, documentation must be wrong. I did a little curling around too and responses of lists just don't look like this snippet. It's always date_changed and date_added. I'll merge this :-)

**Pull request merged**